### PR TITLE
Added infrastructure_encryption to azurerm_storage_account

### PR DESCRIPTION
**Pull request by scottambroseio**
Resolves https://github.com/hashicorp/terraform-provider-azurerm/issues/8816

I've run the new tests in my own subscription and they pass 👍

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
We have a test failure here:
```
------- Stdout: -------
=== RUN   TestAccStorageAccount_storageV1ToV2Update
=== PAUSE TestAccStorageAccount_storageV1ToV2Update
=== CONT  TestAccStorageAccount_storageV1ToV2Update
    testcase.go:110: Step 1/4 error: Error running apply: exit status 1
        
        Error: creating Azure Storage Account "unlikely23exst2acctm4xeb": storage.AccountsClient#Create: Failure sending request: StatusCode=400 -- Original Error: Code="FeatureNotSupportedForAccount" Message="Infrastructure encryption is not supported for the account."
        
          with azurerm_storage_account.test,
          on terraform_plugin_test.tf line 11, in resource "azurerm_storage_account" "test":
          11: resource "azurerm_storage_account" "test" {
        
--- FAIL: TestAccStorageAccount_storageV1ToV2Update (124.02s)
FAIL
```

**Comment by scottambroseio** [comment]
Thanks for letting me know. I ran the new tests I added but not the existing ones 🤦 . This time I ran all the TestAccStorageAccount prefixed tests and they all seem happy 👍 .

File: internal/services/storage/storage_account_resource.go [hunk 1]
```diff
@@ -1096,6 +1108,10 @@ func resourceStorageAccountCreate(d *pluginsdk.ResourceData, meta interface{}) e
 		},
 	}
 
+	if accountKind == string(storage.KindStorageV2) {
+		parameters.Encryption.RequireInfrastructureEncryption = &infrastructureEncryption
+	}
```
Comment (katbyte) [inline internal/services/storage/storage_account_resource.go:1113]: we should probably return an error if the user is trying to set this with v1?

**Comment by katbyte** [comment]
no worries @scottambroseio! tests pass now thou we should probably return an error if the user tried to set this with v1 accounts?

**Review by katbyte — APPROVED** [review]
Thanks @scottambroseio - LGTM now 🏗️ 

However you did commit your test.sh file 😅 so you definitely want to rotate your secrets ASAP!!

**Comment by katbyte** [comment]
added enabled to the property name in #14956

**Pull request merged**